tools/gulp/packaging/inline-resources.ts

```diff
@@ -0,0 +1,58 @@
+/* tslint:disable:no-eval */
+
+import {dirname, join} from 'path';
+import {readFileSync, writeFileSync} from 'fs';
+import {sync as glob} from 'glob';
+
+/** Finds all JavaScript files and inlines all external resources of Angular components. */
+export function inlineResourcesFolder(folderPath: string) {
+  glob(join(folderPath, '**/*.js')).forEach(filePath => inlineResources(filePath));
+}
+
+/** Inlines the external resources of Angular components of a file. */
+export function inlineResources(filePath: string) {
+  let fileContent = readFileSync(filePath, 'utf-8');
+
+  fileContent = inlineTemplate(fileContent, filePath);
+  fileContent = inlineStyles(fileContent, filePath);
+  fileContent = removeModuleId(fileContent);
+
+  writeFileSync(filePath, fileContent, 'utf-8');
+}
+
+/** Inlines the templates of Angular components for a specified source file. */
+function inlineTemplate(fileContent: string, filePath: string) {
+  return fileContent.replace(/templateUrl:\s*'([^']+?\.html)'/g, (match, templateUrl) => {
+    const templateFile = join(dirname(filePath), templateUrl);
+    const templateContent = loadResourceFile(templateFile);
+    return `template: "${templateContent}"`;
+  });
+}
+
+
+/** Inlines the external styles of Angular components for a specified source file. */
+function inlineStyles(fileContent: string, filePath: string) {
+  return fileContent.replace(/styleUrls:\s*(\[[\s\S]*?])/gm, (match, styleUrls) => {
```

Comment:
Maybe rename `styleUrls` to `styleUrlsValue` to make it more clear that it's just yet the real array of styleUrl strings. Then `prsedUrls` below could just change to `styleUrls`.